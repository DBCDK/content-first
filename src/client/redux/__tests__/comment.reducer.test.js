import commentReducer, {
  FETCH_COMMENTS,
  FETCH_COMMENTS_SUCCESS,
  FETCH_COMMENTS_ERROR
} from '../comment.reducer';

const createTestState = () => ({
  expanded: false,
  elements: [],
  isLoading: false,
  pendingMerge: null
});

describe('commentReducer', () => {
  test('Fetch comments', () => {
    const state = createTestState();
    const action = {
      type: FETCH_COMMENTS,
      id: '123'
    };
    const result = commentReducer(state, action);
    expect(result).toMatchSnapshot();
  });
  test('Fetch comments success', () => {
    const state = createTestState();
    const action = {
      type: FETCH_COMMENTS_SUCCESS,
      id: '123',
      comments: [{id: 'comment_1'}]
    };
    const result = commentReducer(state, action);
    expect(result).toMatchSnapshot();
  });
  test('Fetch comments error', () => {
    const state = createTestState();
    const action = {
      type: FETCH_COMMENTS_ERROR,
      id: '123',
      error: 'some error'
    };
    const result = commentReducer(state, action);
    expect(result).toMatchSnapshot();
  });
});
