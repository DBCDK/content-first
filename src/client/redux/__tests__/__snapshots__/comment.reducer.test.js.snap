// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commentReducer Fetch comments 1`] = `
Object {
  "123": Object {
    "loading": true,
  },
  "elements": Array [],
  "expanded": false,
  "isLoading": false,
  "pendingMerge": null,
}
`;

exports[`commentReducer Fetch comments error 1`] = `
Object {
  "123": Object {
    "comments": Array [],
    "error": "some error",
    "loading": false,
  },
  "elements": Array [],
  "expanded": false,
  "isLoading": false,
  "pendingMerge": null,
}
`;

exports[`commentReducer Fetch comments success 1`] = `
Object {
  "123": Object {
    "comments": Array [
      Object {
        "id": "comment_1",
      },
    ],
    "id": "123",
    "loading": false,
  },
  "elements": Array [],
  "expanded": false,
  "isLoading": false,
  "pendingMerge": null,
}
`;
